client/state/support/actions.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SUPPORT_USER_ACTIVATED,
+	SUPPORT_USER_DEACTIVATED,
+	SUPPORT_USER_FETCH_TOKEN,
+} from 'state/action-types';
+
+import User from 'lib/user';
+
+import userSettings from 'lib/user-settings';
+
+/**
+ * Requests a support user token, then dispatches the relevant actions upon response
+ * 
+ * @param  {string} supportUser     Support username
+ * @param  {string} supportPassword Support password
+ * @return {thunk}                  The action thunk
+ */
+export function supportUserFetchToken( supportUser, supportPassword ) {
+	return ( dispatch ) => {
+		if ( !supportUser || !supportPassword ) {
+			return;
+		}
+
+		dispatch( {
+			type: SUPPORT_USER_FETCH_TOKEN,
+			supportUser
+		} );
+
+		const user = new User();
+
+		const activateSupportUser = ( userData ) => 
+			dispatch( supportUserActivated( Object.assign( {}, user.data ) ) );
+
+		const tokenErrorCallback = ( error ) => 
+			dispatch( supportUserDeactivated( error.message ) );
+
+		const changeUserCallback = error => error
+			? tokenErrorCallback( error )
+			: activateSupportUser();
+
+		user.changeUser( supportUser, supportPassword, changeUserCallback, tokenErrorCallback );
+	}
+}
+/**
+ * Logs out of support user, restoring the original user
+ * @return {object} The action object
+ */
+export function supportUserRestore() {
+	let user = new User();
+	user.clear();
+	user.restoreUser();
```

Comment:
There's another issue with these intermediary steps that affect the purity of the state — for part of our offline strategy we are dumping state into storage and hydrating on boot, we are also exploring queues of actions to be replayed later on. Anything that affects what's expected of a pure redux flow would add obstacles in the way. I'm of the mind that as long as we have coexisting data structures we need to duplicate code and handle it in both places, or fully rework the old code to the new state tree.